src/app/config/confparse.c

```diff
@@ -380,6 +380,55 @@ config_new(const config_mgr_t *mgr)
   return opts;
 }
 
+/**
+ * Helper: as config_validate(), but validate only a single format/object
+ * pair.
+*/
+static int
+config_validate_single(const config_format_t *fmt,
+                       const void *oldval,
+                       void *newval,
+                       char **msg_out)
+{
+  if (fmt->validate_fn) {
+    return fmt->validate_fn(oldval, newval, msg_out);
+  } else {
+    return 0;
+  }
+}
+
+/** Use <b>mgr</b> to check whether <b>newval</b> is a valid configuration.
+ * If <b>oldval</b> is present, check whether the transition from
+ * <b>oldval</b> to <b>newval</b> is valid.
+ *
+ * Some of the methods called here may update the values in <b>newval</b>,
+ * though in general they should only update derived fields rather than fields
+ * managed by a configuration format.
```

Comment:
It would be nice if we could split derived and base options, and make the base options read-only.
That would help us avoid a whole class of bugs.

This is a nice to have for #29211.